server/remote_cache/content_addressable_storage_server/content_addressable_storage_server.go

```diff
@@ -154,11 +154,15 @@ func (s *ContentAddressableStorageServer) BatchUpdateBlobs(ctx context.Context,
 		return nil, err
 	}
 
-	canWrite, err := capabilities.IsGranted(ctx, s.env, akpb.ApiKey_CACHE_WRITE_CAPABILITY)
+	canWriteAll, err := capabilities.IsGranted(ctx, s.env, akpb.ApiKey_CACHE_WRITE_CAPABILITY)
```

Comment:
might be cleaner if you modified capabilities.IsGranted to take both at once? then you could leave the code as-is for both bytestream.go and cas.go.

This might even just work if you did capabilities.IsGranted(ctx, s.env, akpb.ApiKey_CACHE_WRITE_CAPABILITY | akpb.ApiKey_CAS_WRITE_CAPABILITY)?